etcdctlv3/README.md

```diff
@@ -109,6 +109,64 @@ OK
 ./etcdctl range foo
 ```
 
+### TXN [options]
+
+TXN applies multiple etcd requests as a single atomic transaction. A transaction consists of list of conditions, a list of requests to apply if all the conditions are true, and a list of requests to apply if any condition is false.
+
+#### Options
+
+- hex -- print out keys and values as hex encoded string
+
+- i -- input transaction with interactive mode
+
+#### Input Format
+Interactive mode:
+```bnf
+<Txn> ::= <CMP>* "\n" <THEN> "\n" <ELSE> "\n"
+<CMP> ::= <key> <CMPOP> <(CMPCREATE|CMPMOD|CMPVAL|CMPVER)> "\n"
+<CMPOP> ::= "g" | "greater" | "e" | "equal" | "l" | "less"
+<CMPCREATE> := ("c"|"create") <revision>
+<CMPMOD> ::= ("m"|"mod") <revision>
+<CMPVAL> ::= ("val"|"value") <value>
+<CMPVER> ::= ("ver"|"version") <version>
+<THEN> ::= <OP>*
+<ELSE> ::= <OP>*
+<OP> ::= ((see put, get, del etcdctl command syntax)) "\n"
+```
+
+TODO: non-interactive mode
+
+#### Return value
+
+Simple reply
+
+- SUCCESS if etcd processed the transaction success list, FAILURE if etcd processed the transaction failure list.
+
+- Simple reply for each command executed request list.
+
+- Additional error string if TXN failed. Exit code is non-zero.
+
+TODO: probably json and binary encoded proto
+
+#### Examples
+
+``` bash
+./etcdctl txn -i
+key1 greater mod 0
```

Comment:
tried
```
./etcdctlv3 txn --i
entry comparison[key target expected_result compare_value] (end with empty line):
foo mod greater 0

entry success request[method key value(end_range)] (end with empty line):
put foo bar

entry failure request[method key value(end_range)] (end with empty line):
put bar foo

Error:  rpc error: code = 3 desc = "etcdserver: key is not provided"
```

Seem like we missed the key when construct the cmp request for some reason.